snorkel/classification/training/loggers/log_writer.py

```diff
@@ -131,4 +131,4 @@ def write_json(self, dict_to_write: Mapping[str, Any], filename: str) -> None:
 
     def close(self) -> None:
         """Close writer if necessary."""
-        pass
+        self.write_log("log.json")
```

Comment:
naming-wise, this (`.close`) feels like the wrong place to do the actual dump to disk, it feels like some cleanup operation should be happening here instead.

If it's a special case / hard to fix class names, let's make it clear in the comments for now

Reply:
Good point. Refactored close() to cleanup() and added some clarifying docstrings.